src/NuGet.Clients/PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -898,6 +913,10 @@ public void CleanUp()
 
             Model.Context.SourceProvider.PackageSourceProvider.PackageSourcesChanged -= Sources_PackageSourcesChanged;
 
+            // make sure to cancel currently running load or refresh tasks
+            _loadCts?.Cancel();
+            _refreshCts?.Cancel();
```

Comment:
`CancellationTokenSource` is disposable. It should be disposed directly or indirectly.